flocker_bb/builders/flocker_vagrant.py

```diff
@@ -462,12 +462,39 @@ def slave_class(self):
 
 TUTORIAL_DISTRIBUTION = "centos-7"
 
+# FLOC-2437 @tomprince already removed centos-7 in
+# 07de6225a1cc35decaba66062e91cea42c791e11
+#
+# Ubuntu-14.04 should stay because it's a supported client platform.
+# It's documented in:
+#  * http://doc-dev.clusterhq.com/using/installing/index.html#installing-flocker-cli
+#
+# Apparently the only reason for running these tests on both rackspace and
+# aws, is to increase the chances of getting a green client installation build
+# when one of the cloud providers is running into api / resource limits.
+#
+# So what I can do is drop the rackspace provider below, which should reduce
+# the number of requests and the resource usage on our Rackspace cloud.
 CLIENT_INSTALLATION_CONFIGURATIONS = [
     ClientConfiguration(provider=provider, distribution=distribution)
     for provider in ('rackspace', 'aws')
     for distribution in ('ubuntu-14.04', 'ubuntu-15.04')
 ]
-
+# But there's no reason (AFAICS) why `run-client-tests` needs to spin up cloud
+# nodes every time. I think it was done that way due to time constraints and
+# because the client tests were originally developed as an option to the
+# run-acceptance-tests script.
+# So if I'm right about that, I'll create follow up issue to improve the
+# ``run-client-tests`` script. For example:
+#
+# 1. By testing the installation of client packages in Docker containers. This
+#    will allow the tests to be run on any of our general on-demand build
+#    slaves. or
```

Comment:
Considering the extent of our CLI tests and the requirements of the CLI itself, using something lightweight like a Docker container for the tests seems appealing.  But:

  * this should be a follow-up issue, not something implemented now
  * the Docker "Ubuntu 14.04" is again different from the things we're testing now and different from what real users are likely to run on their real computers.  Perhaps even more so than the cloud images we're currently testing - since those are at least running on something that *pretends* to be a computer.  The Docker image doesn't really even do that.  Again, I don't know that these differences make any difference to our software.  But that "I don't know" is probably a good reason to learn more and err on the side of more testing.  So ... we could *add* Docker-based CLI builders rather than using them to replace what we have now?  Or, indeed, as you suggest below, use a VM instead of Docker - which is probably roughly as relevant to our users as the current cloud testing we do.